modules/images/webp-uploads/fallback.js

```diff
@@ -0,0 +1,67 @@
+( function( d ) {
+	window.webpUploadsFallbackWebpImages = function( media ) {
+		for ( var i = 0; i < media.length; i++ ) {
+			try {
+				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
+				if ( ! ext || ! ext[0] ) {
+					continue;
+				}
+
+				var images = d.querySelectorAll( 'img.wp-image-' + media[i].id );
+				for ( var j = 0; j < images.length; j++ ) {
+					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
+					var srcset = images[j].getAttribute( 'srcset' );
+					if ( srcset ) {
+						images[j].setAttribute( 'srcset', srcset.replace( /\.webp(\s)/ig, ext[0] + '$1' ) );
+					}
+				}
+			} catch ( e ) {
+			}
+		}
+	};
+
+	var restApi = d.getElementById( 'webpUploadsFallbackWebpImages' ).getAttribute( 'data-rest-api' );
+
+	var loadMediaDetails = function( nodes ) {
+		var ids = [];
+		for ( var i = 0; i < nodes.length; i++ ) {
+			if ( nodes[i].nodeName !== "IMG" || ! nodes[i].src.match( /\.webp$/i ) ) {
```

Comment:
We probably should also check if the `srcset` includes any `.webp` files? Especially in the context of #372, it is certainly possible that the main `src` would be a JPEG while some images in `srcset` would be WebP and thus qualify for the fallback.

Reply:
Updated.